rasa/core/domain.py

```diff
@@ -729,13 +742,30 @@ def as_dict(self) -> Dict[Text, Any]:
     def persist(self, filename: Union[Text, Path]) -> None:
         """Write domain to a file."""
 
-        domain_data = self.as_dict()
+        domain_data = self.domain_with_intents_transformed_for_file()
         utils.dump_obj_as_yaml_to_file(filename, domain_data)
 
-    def cleaned_domain(self) -> Dict[Text, Any]:
-        """Fetch cleaned domain, replacing redundant keys with default values."""
-
+    def domain_with_intents_transformed_for_file(self) -> Dict[Text, Any]:
+        """Replace the internal `used_entities` property by `use_entities` or `ignore_entities`."""
         domain_data = self.as_dict()
+        for idx, intent_info in enumerate(domain_data["intents"]):
+            for name, intent in intent_info.items():
+                use_entities = set(intent.pop("used_entities"))
+                ignore_entities = set(domain_data["entities"]) - use_entities
+                if len(use_entities) == len(domain_data["entities"]):
+                    intent["use_entities"] = True
+                elif len(use_entities) <= len(domain_data["entities"]) / 2:
+                    intent["use_entities"] = list(use_entities)
+                else:
+                    intent["ignore_entities"] = list(ignore_entities)
```

Comment:
what if you want to use all entities and just exclude a single one? IMO something like `use_entities: True` and `ignore_entities: [entity_1]` should be possible

Reply:
By default, all entities are included except the ones listed in `ignore_entities`. So the option you are asking for already exists.
A domain file with an intent that has `use_entities: True` and `ignore_entities: [entity_1]` is equivalent to one that only has `ignore_entities: [entity_1]` and no field `use_entities`. For writing back into a file, the "simplest" variant is chosen, so the unnecessary field is simply left out.

Reply:
👍